### PR TITLE
feat(router): allow operation name propagation

### PR DESCRIPTION
**Pull request by Noroth**
## Motivation and Context

This PR adds an option to propagate the graphql operation name to the corresponding subgraph calls.

## Related Engine Release

https://github.com/wundergraph/graphql-go-tools/releases/tag/v2.0.0-rc.128

## Checklist

- [ ] I have discussed my proposed changes in an issue and have received approval to proceed.
- [ ] I have followed the coding standards of the project.
- [ ] Tests or benchmarks have been added or updated.
- [ ] Documentation has been updated on [https://github.com/wundergraph/cosmo-docs](https://github.com/wundergraph/cosmo-docs).
- [ ] I have read the [Contributors Guide](https://github.com/wundergraph/cosmo/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: router/pkg/config/config.go [hunk 1]
```diff
@@ -320,6 +320,7 @@ type EngineExecutionConfiguration struct {
 	EnableValidationCache                  bool                     `envDefault:"true" env:"ENGINE_ENABLE_VALIDATION_CACHE" yaml:"enable_validation_cache"`
 	ValidationCacheSize                    int64                    `envDefault:"10240" env:"ENGINE_VALIDATION_CACHE_SIZE" yaml:"validation_cache_size,omitempty"`
 	ResolverMaxRecyclableParserSize        int                      `envDefault:"32768" env:"ENGINE_RESOLVER_MAX_RECYCLABLE_PARSER_SIZE" yaml:"resolver_max_recyclable_parser_size,omitempty"`
+	EnableOperationNamePropagation         bool                     `envDefault:"true" env:"ENGINE_ENABLE_OPERATION_NAME_PROPAGATION" yaml:"enable_operation_name_propagation"`
```
Comment (StarpTech) [inline router/pkg/config/config.go:323]: We should be much more precise in the naming. For example `EnableSubgraphFetchOperationName`

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router/pkg/config/config.go (hunk 1 above)
Comment (Noroth) [reply router/pkg/config/config.go]: tried to be more precise with naming and description in the schema

File: router-tests/query_plans_test.go [hunk 2]
```diff
@@ -168,4 +168,30 @@ func TestQueryPlans(t *testing.T) {
 			}
 		})
 	})
+	t.Run("include operation name in each request", func(t *testing.T) {
+		testenv.Run(t, &testenv.Config{
+			ModifyEngineExecutionConfiguration: func(cfg *config.EngineExecutionConfiguration) {
+				cfg.Debug.AlwaysIncludeQueryPlan = true
+				cfg.EnableOperationNameToSubgraphPropagation = true
```
Comment (StarpTech) [inline router-tests/query_plans_test.go:175]: I still think this name over complicates it 👀 .

`EnableFetchOperationName` 
`EnableSubgraphFetchOperationName `

?

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**